Add main functionality

illiapelypenko opened the pull request.

Commit:
In development

Commit:
Add SearchResults & PropertyList & PropertyCard Delete extra file

Commit:
Fix obboarding animation
Change assets naming
Add header to development

Commit:
Fix onboarding animation
Change assets naming
Refactor TabNavigator

illiapelypenko committed:
Fix PR issues

illiapelypenko committed:
Refactor PropertyCard
Add RootNavigator header styling

illiapelypenko committed:
Merge branch 'search-screen' into search-results

illiapelypenko committed:
In development

illiapelypenko committed:
Refactor TabNavigator

Commit:
In development

Commit:
Style PorpertyCard

illiapelypenko committed:
Add search-results
Add search-page
Refactor PropertyCard

illiapelypenko committed:
Add search-results
Add search-page
Refactor code

Commit:
Add code refactor

Commit:
Fix IOS PropertyCard styles

Commit:
Fix PR issues

Commit:
Fix PR issues

illiapelypenko committed:
Add error handling
Add async storage

illiapelypenko committed:
Add error handling
Add async storage

uladzislau-stuk commented on App.js:
```diff
@@ -1,20 +1,39 @@
+    if (favorites) dispatch(setFavorites(favorites));
+  };
+
+  return <RootNavigation alreadyLaunched={alreadyLaunched} />;
```
It'd better to pass parameter named as initialRouteName

uladzislau-stuk commented on src/components/SizeButton.js:
```diff
@@ -0,0 +1,32 @@
+  );
+};
+
+const styles = StyleSheet.create({
+  container: {
+    position: 'absolute',
+    right: 0,
+    padding: 20,
+  },
+});
```
https://en.bem.info/methodology/css/#external-geometry-and-positioning

uladzislau-stuk commented on src/screens/Favorites.js:
```diff
@@ -1,12 +1,45 @@
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    navigation.addListener('focus', checkFavoritesLength);
+  }, []);
```
https://reactnavigation.org/docs/use-focus-effect

uladzislau-stuk commented on src/screens/Onboarding.js:
```diff
@@ -34,45 +36,53 @@ const slides = [
   const [activeSlide, setActiveSlide] = useState(0);
   const [prevActiveSlide, setPrevActiveSlide] = useState(null);
 
+  const alreadyLaunched = useSelector(state => state.alreadyLaunched);
+
+  useEffect(() => {
+    if (alreadyLaunched) navigation.navigate(ROUTES.TAB_NAVIGATOR);
+  }, []);
```
You don't need it

uladzislau-stuk commented on src/navigators/RootNavigator.js:
```diff
@@ -1,23 +1,108 @@
+    }
+  }, [error.message]);
+
+  useEffect(() => {
+    if (alreadyLaunched) {
+      ref.current.navigate(ROUTES.TAB_NAVIGATOR);
+    }
+  }, [alreadyLaunched]);
```
You can remove it if you pass prop initialRouteName (the name of the route to render on first load of the navigator.)
https://reactnavigation.org/docs/stack-navigator

Commit:
Add default StatusBar
Remove SizeButton styles
Fix android styles in PropertyCards & PropertyScreen
Change show onboarding screen logic
Add headerRight styles
Change headerTitleContainerStyle
Fix ErrorScreen & Onboarding & Search styles
Refactor Favotites screen logic

Commit:
Add default StatusBar
Remove SizeButton styles
Fix android styles in PropertyCards & PropertyScreen
Change show onboarding screen logic
Add headerRight styles
Change headerTitleContainerStyle
Fix ErrorScreen & Onboarding & Search styles
Refactor Favotites screen logic

illiapelypenko committed:
Fix IOS styles
Fix textInput styles in Search
Fix headerTitleAlign in RootNavigation
Fix padding on PropertyCardMinified

illiapelypenko committed:
Fix IOS styles
Fix textInput styles in Search
Fix headerTitleAlign in RootNavigation
Fix padding on PropertyCardMinified

illiapelypenko committed:
Fix ios header styles

illiapelypenko committed:
Fix ios header styles

illiapelypenko committed:
Move SearchButton logic to separate file

illiapelypenko committed:
Move SearchButton logic to separate file

illiapelypenko committed:
Merge branch 'search-results' of https://github.com/illiapelypenko/nativehomefind into search-results

illiapelypenko committed:
Add removeEventListener to Search
